main.nf

```diff
@@ -21,56 +22,54 @@ params.LD_BLOCKS = "${projectDir}/assets/NO_LD_BLOCKS"
 params.FLASHPCA_EXCLUSION_REGIONS = "${projectDir}/assets/exclusion_regions_hg19.txt"
 
 // Estimands Generation 
-params.BATCH_SIZE = 400
+params.BATCH_SIZE = 50
 
 // CUSTOM
-params.STUDY_DESIGN = "CUSTOM"
-params.ESTIMANDS_FILE = "NO_ESTIMANDS_FILE"
-
-// FROM_ACTORS
-params.EXTRA_CONFOUNDERS = "${projectDir}/assets/NO_EXTRA_CONFOUNDER"
-params.EXTRA_COVARIATES = "${projectDir}/assets/NO_EXTRA_COVARIATE"
-params.ENVIRONMENTALS = "${projectDir}/assets/NO_EXTRA_TREATMENT"
-params.ORDERS = "1,2"
+params.ESTIMANDS_CONFIG = "NO_ESTIMANDS_CONFIG"
 
 // SVP Parameters
 params.SVP = false
 params.GRM_NSPLITS = 100
 params.NB_SVP_ESTIMATORS = 100
 params.MAX_SVP_THRESHOLD = 0.9
-params.ESTIMATOR_KEY = "TMLE"
+params.ESTIMATOR_KEY = "1"
 
 // TMLE Parameters
 params.KEEP_IC = params.SVP == true ? true : false
 params.PVAL_THRESHOLD = 0.05
-params.TL_SAVE_EVERY = 100
-params.ESTIMATOR_FILE = "glmnet"
-
-// Outputs Parameters
-params.ARROW_OUTPUT = "dataset.arrow"
-params.JSON_OUTPUT = "NO_JSON_OUTPUT"
-params.HDF5_OUTPUT = "results.hdf5"
+params.TL_SAVE_EVERY = params.BATCH_SIZE
+params.ESTIMATORS_CONFIG = "wtmle-ose--tunedxgboost"
 
-// Negative Control Parameters
-params.PERMUTATION_HDF5_OUTPUT = "permutation_${params.HDF5_OUTPUT}"
-params.PERMUTATION_JSON_OUTPUT = params.JSON_OUTPUT == "NO_JSON_OUTPUT" ? params.JSON_OUTPUT : "permutation_${params.JSON_OUTPUT}"
-params.MAX_PERMUTATION_TESTS = ""
-params.MAX_PERMUTATION_ATTEMPTS = 1
-params.PERMUTATION_ORDERS = "1"
-params.MAF_MATCHING_RELTOL = 0.05
-params.VARIANTS_TO_RANDOMIZE = "NO_VARIANT_TO_RANDOMIZE"
-params.N_RANDOM_VARIANTS = 10
+// Simulations
+params.TRAIN_RATIO = 6
+params.SAMPLE_GA_HITS = true
+params.GA_MAX_VARIANTS = 50
+params.GA_DISTANCE_THRESHOLD = 1000000
+params.GA_PVAL_THRESHOLD = 1e-5
+params.GA_MAF_THRESHOLD = 0.01
+params.N_REPEATS = 2
+params.RNGS = 1..250
+params.SAMPLE_SIZES = [500000]
+params.GA_TRAIT_TABLE = "${projectDir}/assets/Traits_Table_GeneATLAS.csv"
+params.MIN_FACTOR_LEVEL_OCCURENCES = 10
+params.MAX_SAMPLING_ATTEMPTS = 1000
+params.NSAMPLES_FOR_TRUTH = 1000000
 
+include { GWAS } from './workflows/gwas.nf'
 include { TARGENE } from './workflows/targene.nf'
-include { PERMUTATION_TEST; RANDOMIZATION_TEST } from './workflows/negative_control.nf'
 include { PCA } from './workflows/pca.nf'
 include { MAKE_DATASET } from './workflows/dataset.nf'
+include { NULL_SIMULATION; REALISTIC_SIMULATION } from './workflows/simulations.nf'
+
+def isGWAS(){
+    config = new Yaml().load(new FileReader(params.ESTIMANDS_CONFIG))
+    return config.type == 'gwas'
+}
```

Comment:
Could use this function to also define some kind of default value for $params.BGEN_FILES so it doesn't throw a warning when you run the gwas mode, if you do not provide BGEN_FILES?

Reply:
what would be natural defaults for BGEN_FILES?

Reply:
probably similar to other cases, "NO_BGEN_FILES" like you have for "NO_QC_FILE" etc.

Reply:
Yes the problem with that is the error message thrown by nextflow will be as follows when the files are actually required.

Reply:
we could have custom error handling based on provided inputs but this is beyond this PR I think.

Reply:
Fair enough!